### ISSUE TITLE
Checking Validation Status inside onChange Handler

### ISSUE THREAD

**Issue by param-fk** [erikras/redux-form#3682]
### Are you submitting a **bug report** or a **feature request**?

**bug report**

### What is the current behavior?

I'm trying to check the validity status of the form on the updation of any field of my form. I'm using onChange handler for that like this
```
const InfoForm = reduxForm({
    form: 'infoTabFormSD',
    onChange: (values, dispatch, props: any) => {
        console.log(props.valid && props.dirty);
        props.setSaveReportBadge(props.valid && props.dirty);
    }
});
```
My fields have field level validation applied like required and minLength etc. 
The problem is these values aren't giving me the right real time result. The form is valid even when I make the required textfield blank. It only updates the valid status when I blur the textfield or type soemthing in the other input field.

### What is the expected behavior?
It should give the real time updated validation status of the form in the `onChange` handler.

### What's your environment?

"redux-form": "^6.0.5",
OS: Mac OS X
Node Version - 7.10.1

### Other information

![screen shot 2017-12-11 at 8 26 26 pm](https://user-images.githubusercontent.com/33153033/33837199-9ed8cd32-deb1-11e7-96e9-72d570aed66c.png)

**Comment by gustavohenke** [comment]
That's a won't fix - the `onChange` handler can prevent the value to make it into the form state (by calling `event.preventDefault()`), therefore this needs to happen before the validation runs.
Also I don't believe flexibilising Redux Form to allow such behaviour would be feasible.

**Comment by param-fk** [comment]
whats the way around then for such a use case? @gustavohenke @

**Comment by gustavohenke** [comment]
Oh, now I see you were talking about the form-wide `onChange` handler.
In this case forget about the `event.preventDefault()` part I mentioned above, I thought we were talking about field-level events.

Anyway, I don't know if we have a better way other than reusing the validate function:

```
reduxForm({
  form: 'bla',
  validate: validateBla,
  onChange (values) {
    const isValid = Object.keys(validateBla(values)).length > 0;
  }
})
```

**Comment by param-fk** [comment]
I finally got it working through this:
```
componentWillReceiveProps(nextProps: IInfoProps) {
        if (nextProps.valid !== this.props.valid || nextProps.dirty !== this.props.dirty) {
            this.props.setSaveReportBadge(nextProps.valid && nextProps.dirty);
        }
    }
```

**Comment by gustavohenke** [comment]
I'm glad you found a workaround!
